pkg/apis/pipeline/v1beta1/resource_types.go

```diff
@@ -129,7 +129,7 @@ type PipelineResourceBinding struct {
 // PipelineResourceResult used to export the image name and digest as json
 type PipelineResourceResult struct {
 	Key          string `json:"key"`
-	Value        string `json:"value"`
+	Value        ArrayOrString `json:"value"`
 	ResourceName string `json:"resourceName,omitempty"`
```

Comment:
As suggested by @ywluogg ,  we can also to this to separate the usage of result value and other value, since in other cases there is no need to support array. 
```suggestion
	Value        String `json:"value"`
	ResultValue *ArrayOrString `json:"resultValue,omitempty"`
	ResourceName string `json:"resourceName,omitempty"`
```

I'd like to provide this solution first and see how other ppl think of this

Reply:
Option 1:  Change Value type from String to ArrayOrString
Pro: backwards compatible without reducing termination msg limit
Con: In some other places we only need String and doesn’t necessarily need to use ArrayOrString

Option 2: Add a new field `ResultValue *ArrayOrString`
Pro: Separate the usage of emitting results (where we need array) and other data(where we only need string) to termination path
Con: reduce the available termination msg size